faucet/src/faucet.rs

```diff
@@ -261,7 +266,7 @@ impl Faucet {
 
         for (balance, client) in clients {
             // Fund all clients who have less than average balance.
-            if balance < average_balance {
+            if options.enable_funding && balance < average_balance {
```

Comment:
This should really be a percentage  (like 80 or 90 percent) of the average balance. There will always be balances under the average unless every balance is exactly the same.

Reply:
I mean my thinking was since it only runs on startup this is fine. I will try to do a proper fix coming week, for now just wanted to make it possible to not run this code but still run the faucet.